lib/Alchemy/Phrasea/Command/Record/RescanFilesMetadata.php

```diff
@@ -0,0 +1,339 @@
+            $output->writeln("<error> wrong value for --multi, use replace or merge</error>");
+
+            return 0;
+        }
+
+        $maxRecord = $input->getOption("max_record_id");
+        $minRecord = $input->getOption("min_record_id");
+
+        if ($input->getOption("databox") != NULL) {
+            try {
+                $db = $this->getDatabox($input->getOption("databox"));
+                if ($db == null) {
+                    $output->writeln("<error>databox not found</error>");
+
+                    return 0;
+                }
+                $dboxes = [$db];
+            } catch (\Exception $e) {
+                $dboxes = [];
+            }
+        } else {
+            $dboxes = $this->appbox->get_databoxes();
+        }
+
+        //  sql request
+        $clauses[]= '1';
+        if ($minRecord != NULL) {
+            $clauses[] = "record_id >= " . $minRecord;
+        }
+
+        if ($maxRecord != NULL) {
+            $clauses[] = "record_id <= " . $maxRecord;
+        }
+
+        if ($type != null) {
+            $clauses[] = "type = " . $type;
+        }
+
+        if($partitionCount !== null && $partitionIndex !== null) {
+            $clauses[] = " MOD(`record_id`, " . $partitionCount . ")=" . ($partitionIndex-1);
+        }
+
+        ///
+
+        $this->unicode = new \unicode();
+
+        foreach ($dboxes as $databox) {
+            $sbasId = $databox->get_sbas_id();
+            $sbasName = $databox->get_dbname();
+            $rToDoCount = $rDoneCount = $rTagFound = 0;
+
+            $field = $this->getField($sbasId, $destination);
+
+            if ($field == null) {
+                $output->writeln(sprintf("<error>Field %s not found on database %s </error>", $destination, $sbasName));
+                continue;
+            }
+
+            if ( $col != null) {
+                $collection = $this->getCollection($sbasId, $col);
+                if ($collection == null) {
+                    $output->writeln(sprintf("<error>collection %s not found on database %s </error>", $col, $sbasName));
+                    continue;
+                }
+
+                $collId = $collection->get_coll_id();
+
+                $clauses[] = "coll_id = " . $collId;
+            }
+
+            if ($source == null) {
+                // use the destination field src
+                $sourceTag = $field->get_original_source();
+            }
+
+            $metaStructId = $field->get_id();
+            $fieldName = $field->get_name();
+            $fieldType = $field->get_type();
+
+            $action = "set";
+
+            if ($field->is_multi()) {
+                if ($multi != 'replace') {
+                    $action = 'add';
+                }
+            }
+
+            $output->writeln(sprintf("<comment>Working on database %s with Id : %d</comment>", $sbasName, $sbasId));
+
+
+            $sql_where = join(" AND ", $clauses);
+
+            $sql = "SELECT * FROM record WHERE " . $sql_where;
```

Comment:
don't use * ;
**overwrite = null** optim: `... left join metadata... where ... and isnull(metadata)` so the select will not fetch records with value(s).